### PR TITLE
Fix 8dof

### PR DESCRIPTION
**Pull request by BenStarmerSmith**
## Proposed changes

Fixed issue where it would create the icons even if conditional was not defined.

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added automated tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [x] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [ ] Manually tested on hardware (if hardware specific or related).

## Documentation

- [x] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

File: ansible/roles/products/teleop/server/desktop-icons/tasks/real-robots.yml [hunk 1]
```diff
@@ -268,7 +268,7 @@
     launch_file_input: "{{ launch_file_input_gui_right_realhw_8DOF }}"
     desktop_icon_name: "4 - Launch Right Teleop GUI 8DOF"
     desktop_icon_path: "Shadow Advanced Launchers/4 - Launch Right Teleop GUI 8DOF"
-  when: bimanual|bool or hand_side=="right" and arm_servo|bool == true
+  when: (bimanual|bool or hand_side=="right") and (arm_servo|bool)
```
Comment (maxzieba) [inline ansible/roles/products/teleop/server/desktop-icons/tasks/real-robots.yml:271]: I think the convention we use is 
- condition1
- condition2

File: ansible/roles/products/teleop/server/desktop-icons/tasks/only-simulation.yml [hunk 2]
```diff
@@ -189,7 +189,7 @@
     desktop_icon_path: "Shadow Advanced Launchers/3 - Launch Right Teleop Simulation 8DOF"
     project_name_input: "{{ project_name_input_value }}"
     launch_file_input: "{{ launch_file_input_gui_right_8DOF }}"
-  when: arm_servo|bool == true
+  when: arm_servo|bool
```
Comment (maxzieba) [inline ansible/roles/products/teleop/server/desktop-icons/tasks/only-simulation.yml:192]: Does this have to be defined in the one-liner or is it optional?

File: ansible/roles/products/teleop/server/desktop-icons/tasks/real-robots.yml (hunk 1 above)
Comment (BenStarmerSmith) [reply ansible/roles/products/teleop/server/desktop-icons/tasks/real-robots.yml:271]: We use both, either one is fine, I think we mainly use this for 3+ conditionals.

File: ansible/roles/products/teleop/server/desktop-icons/tasks/only-simulation.yml (hunk 2 above)
Comment (BenStarmerSmith) [reply ansible/roles/products/teleop/server/desktop-icons/tasks/only-simulation.yml:192]: Its optional, false by default, so you have to enable it to work.

**Review by maxzieba — APPROVED** [review]

**Review by biofotis — APPROVED** [review]

**Pull request merged**